naga-cli/src/bin/naga.rs

```diff
@@ -487,48 +594,55 @@ fn parse_input(
                 }
             }
         }
-        ext @ ("vert" | "frag" | "comp" | "glsl") => {
+        InputKind::Glsl => {
+            let shader_stage = match params.shader_stage {
+                Some(shader_stage) => shader_stage,
+                None => ShaderStage::from_str(
```

Comment:
Once you have `from_str` defined, you can use `my_string.parse()` with return type deduction.

Reply:
This doesn't seem to have been addressed?

Reply:
Oops. I think I rebased it back onto an old version of the branch when updating to the current trunk :confused:

Reply:
Ah yeah, it happens :)